substrate/frame/assets/src/traits.rs

```diff
@@ -0,0 +1,19 @@
+pub mod sufficients {
+	use sp_runtime::DispatchResult;
+
+	/// Trait for providing the sufficient state of an asset.
+	pub trait Inspect<AssetId> {
+		/// Returns whether an asset is sufficient or not
+		fn is_sufficient(asset_id: AssetId) -> bool;
+	}
+
+	/// Trait for mutating the sufficient state of an asset
+	pub trait Mutate<AssetId> {
+		/// Sets the `is_sufficient` value of an asset.
+		///
+		/// ### Errors
+		///
+		/// - [`Unknown`][crate::Error::Unknown] when the asset ID is unknown.
+		fn set_sufficient(asset_id: AssetId, is_sufficient: bool) -> DispatchResult;
```

Comment:
I would either name this as such or split into two functions: `make_sufficient(asset_id)` and `make_insufficient(asset_id)`.
```suggestion
		fn set_sufficiency(asset_id: AssetId, is_sufficient: bool) -> DispatchResult;
```

Reply:
I like the `make_sufficient`/`make_insufficient` approach better. While a bit verbose now, it might fit better in a situation where something is needed to be done on each change (i.e. should anything happen when making a sufficient asset insufficient?).